### DISCUSSION TITLE
Escaping query params in textDirectives [remark-directive]

### DISCUSSION THREAD

**Discussion by Lambdaderuta** [remarkjs/remark#913] [Q&A] [answered]
Hi!
Didn't find an answer to my question so...
I'm trying to put some link in textDerictive via  a:[link]{someParams='foo' url='//:example.com'} and it's works
But when link has query params in example youtube with ?v=foobar it's breaks cause of '=' sign.
Is there a way to avoid this behavior? Some kind of escape? Tried '\\' before '=' but it not worked.

**Comment by wooorm** [comment, marked as answer]
Can you use markdown code to show your problem? I don’t know what you’re asking. What is breaking? What error? What are you using?

**Reply by ChristianMurphy** [reply]
I'd second that question, pulling together an example based off my understanding of the question the initial example doesn't make sense https://stackblitz.com/edit/node-twvttn?file=index.js :thinking:

**Comment by parkyoung555** [comment]
Hey all!

If I'm understanding the original question correctly, I believe I'm running into the same, or similar issue: Instances of HTML entities within directive attribute values are unexpectedly being converted the equivalent special character.

Example directive:
```
:link[link]{someParams='foo' url='https://www.google.com?t=234&test=hello&param=90'}
```

The resulting `url` attribute value is `https://www.google.com?t=234&test=hello¶m=90`. This threw me off guard a bit since I did not expect `&para` to evaluate to `¶` without the terminating semicolon.

Code sandbox:
https://stackblitz.com/edit/node-24nvom?file=index.js

Update:
It looks like in version `^1.0.0` of remark-directive the HTML entity decoding only occurs if the terminating semicolon is present (e.g. `https://www.google.com?t=234&test=hello&para;m=90`), which seems to be the correct behavior.

**Reply by wooorm** [reply]
Not needing a closing semicolon on several character references is how HTML works:

```js
document.body.innerHTML = 'https://www.google.com?t=234&test=hello&param=90'; console.log(document.body.innerHTML)
```
```txt
https://www.google.com?t=234&amp;test=hello¶m=90
```

Needing a closing semicolon is how markdown works in most places.
In HTML, character references do work slightly different in attributes:

```js
document.body.innerHTML = '<a href="https://www.google.com?t=234&test=hello&param=90">x</a>'; console.log(document.body.innerHTML)
```
```txt
<a href="https://www.google.com?t=234&amp;test=hello&amp;param=90">x</a>
```

I think in this case it makes sense to stick close to how markdown works and close to how attributes in HTML work.

**Reply by wooorm** [reply]
Fixed in https://github.com/micromark/micromark-extension-directive/releases/tag/2.1.2 and https://github.com/syntax-tree/mdast-util-directive/releases/tag/2.2.2. Thanks for clarifying the question!
